### PR TITLE
Add error page feature to Traefik

### PR DESCRIPTION
**Pull request by bparli**
To address [JIRA 3328](https://jira.gonitrocorp.com/browse/ENG-3328) and since the custom error page isn't an existing Traefik feature
- Add ErrorPages handler for serving custom error page
- Add error page handler to frontend forwarder
- Add error page handler to backend loadbalancer
- expose optional configuration in traefik.toml
- Unit tests to ensure page is being served correctly 

@felixgborrego @relistan @mihaitodor Do you guys mind having a look?

### CODE REVIEW

File: configuration.go [hunk 1]
```diff
@@ -24,22 +25,23 @@ type TraefikConfiguration struct {
...
+	ACME                      *acme.ACME         `description:"Enable ACME (Let's Encrypt): automatic SSL"`
+	DefaultEntryPoints        DefaultEntryPoints `description:"Entrypoints to be used by frontends that do not specify any entrypoint"`
+	ProvidersThrottleDuration flaeg.Duration     `description:"Backends throttle duration: minimum duration between 2 events from providers before applying a new configuration. It avoids unnecessary reloads if multiples events are sent in a short amount of time."`
+	MaxIdleConnsPerHost       int                `description:"If non-zero, controls the maximum idle (keep-alive) to keep per-host.  If zero, DefaultMaxIdleConnsPerHost is used"`
+	IdleTimeout               flaeg.Duration     `description:"maximum amount of time an idle (keep-alive) connection will remain idle before closing itself."`
+	InsecureSkipVerify        bool               `description:"Disable SSL certificate verification"`
+	Retry                     *Retry             `description:"Enable retry sending request if network error"`
+	ErrorPages                *middlewares.ErrorPages
```
Comment (relistan) [inline configuration.go]: Looks like this needs a `description` struct tag entry like the others.

**Comment by relistan** [comment]
Looks solid to me. Nice @bparli !

**Review by mihaitodor — APPROVED** [review]
LGTM

File: configuration.go (hunk 1 above)
Comment (bparli) [reply configuration.go]: Thanks for catching this @relistan

**Pull request merged**